lib/jsonpath/parser.rb

```diff
@@ -8,20 +8,23 @@ class JsonPath
   class Parser
     def initialize(node)
       @_current_node = node
+      @_expr_map = {}
     end
 
     def parse(exp)
       exps = exp.split(/(&&)|(\|\|)/)
-      ret = parse_exp(exps.shift)
+      construct_expression_map(exps)
+      @_expr_map.each {|k, v| exp.sub!(k, "#{v}")}
+      eval(exp)
```

Comment:
This eval never parses real code. It only parses something like:
`"((false || false) && (false || true))"`

Never the less I will replace it with my own quick parser.